sklearn/linear_model/logistic.py

```diff
@@ -1332,9 +1336,13 @@ def predict_proba(self, X):
         """
         if not hasattr(self, "coef_"):
             raise NotFittedError("Call fit before prediction")
-        calculate_ovr = self.coef_.shape[0] == 1 or self.multi_class == "ovr"
-        if calculate_ovr:
+        if self.multi_class == "ovr":
             return super(LogisticRegression, self)._predict_proba_lr(X)
+        elif self.coef_.shape[0] == 1:
```

Comment:
I find it slightly clearer to check the shape of the decision function, i.e. something like:

```py
else:
    decision == self.decision_function(X)
    decision_2d = decision if decision.ndim > 1 else np.c_[-decision, decision]
    return softmax(decision_2d, copy=False)
```